.github/workflows/CI.yml

```diff
@@ -16,16 +18,20 @@ jobs:
       with:
         python-version: ${{ matrix.python-version }}
         architecture: x64
+    - name: Run tests
+      shell: bash
+      run: |
+        python -m venv venv
+        source venv/bin/activate
+        pip install -U pip
+        pip install -r build.requirements.txt
+        maturin develop
+        python -m pytest
     - uses: messense/maturin-action@v1
       with:
         manylinux: auto
         command: build
         args: --release -o dist --interpreter python${{ matrix.python-version }}
-    - name: Run tests
-      run: |
-        pip install bytewax --find-links dist -v --pre
-        pip install --upgrade pytest==7.0.0
-        python -m pytest pytests/
     - name: Upload wheels
```

Comment:
Does this upload wheels at all on PR runs? Should that only happen on tagged releases?